src/test/unitTests.js

```diff
@@ -0,0 +1,139 @@
+const { describe, it, beforeEach, afterEach } = require('mocha')
+const axios = require('axios')
+const chai = require('chai')
+const sinon = require('sinon')
+const UserDataHandler = require('../data_handlers/user_data_handler')
+
+let sandbox
+let userDataHandler
+const expect = chai.expect
+
+describe('UserDataHandler', () => {
+  beforeEach(() => {
+    sandbox = sinon.createSandbox()
+    userDataHandler = new UserDataHandler()
+  })
+
+  afterEach(() => {
+    sandbox.restore()
+  })
+
+  it('should load user data', async () => {
+    const data = [{ id: 1, email: 'test1@gmail.com' }, { id: 2, email: 'test2@gmail.com' }]
+    sandbox.stub(axios, 'get').returns(Promise.resolve({ data }))
+
+    await userDataHandler.loadUsers()
+    expect(userDataHandler.users.length).to.equal(2)
+  })
+
+  it('should return user emails list', () => {
+    userDataHandler.users = [{ id: 1, email: 'test1@gmail.com' }, { id: 2, email: 'test2@gmail.com' }]
+
+    const emails = userDataHandler.getUserEmailsList()
+    expect(emails).to.equal('test1@gmail.com;test2@gmail.com')
+  })
+
+  it('should return number of users', () => {
+    userDataHandler.users = [{ id: 1, email: 'test1@gmail.com' }, { id: 2, email: 'test2@gmail.com' }]
```

Comment:
this test depends on the implementation, not on the interface. use loadUsers() method instead